slurm_deploy/deploy_ctl.sh

```diff
@@ -80,104 +71,119 @@ function item_download() {
             fi
         else
-            create_dir $SRC_DIR/$REPO_NAME
-            fname=`basename $packurl`
-            is_gzip=`echo $fname | grep "tar\.gz"`
-            is_bzip=`echo $fname | grep "tar\.bz"`
+            create_dir "$SRC_DIR/$REPO_NAME"
+            fname=$(basename "$packurl")
+            is_gzip=$(echo "$fname" | grep "tar\.gz")
+            is_bzip=$(echo "$fname" | grep "tar\.bz")
             tar_opts=""
             if [ -n "$is_gzip" ]; then
                 tar_opts="-xz"
             elif [ -n "$is_bzip" ]; then
                 tar_opts="-xj"
-        	else
+            else
                 echo_error $LINENO "\"$REPO_NAME\" Repository can not be obtained: Unknown archive type: $fname, only .gz and .bz2 are supported"
-                rm -rf $SRC_DIR/$REPO_NAME
+                rm -rf "${SRC_DIR:?}/$REPO_NAME"
                 exit 1
-        	fi
-        	echo "tar_opts = $tar_opts"
-            if [ ${packurl:0:1} = "/" ]; then
+            fi
+            echo "tar_opts = $tar_opts"
+            if [ "${packurl:0:1}" = "/" ]; then
                 echo "unpacking \"$REPO_NAME\" from local path..."
-                cat $packurl | tar $tar_opts -C $SRC_DIR/$REPO_NAME --strip-components 1
+                # TODO
+                # shellcheck disable=SC2002
+                cat "$packurl" | tar $tar_opts -C "$SRC_DIR/$REPO_NAME" --strip-components 1
             else
-                curl -L $packurl | tar $tar_opts -C $SRC_DIR/$REPO_NAME --strip-components 1
+                curl -L "$packurl" | tar $tar_opts -C "$SRC_DIR/$REPO_NAME" --strip-components 1
             fi
             if [ "0" -ne "${PIPESTATUS[0]}" ]; then
                 echo_error $LINENO "\"$REPO_NAME\" Repository can not be obtained. Cannot continue. "
-                rm -rf $SRC_DIR/$REPO_NAME
+                rm -rf "${SRC_DIR:?}/$REPO_NAME"
                 exit 1
             fi
         fi
-        
+
         if [ -n "$commit" ]; then
-            cd $REPO_SRC
-            git checkout -b test $commit
+            cd "$REPO_SRC"
+            git checkout -b test "$commit"
             cd -
         fi
     fi
 
+    # shellcheck disable=SC2181
     if [ "$?" != "0" ]; then
         echo_error $LINENO "\"$REPO_NAME\": Repository can not be prepared. Cannot continue."
         exit 1
     fi
 
     build=$REPO_NAME/.build
     if [ ! -d "$build" ]; then
-        create_dir $build
+        create_dir "$build"
     fi
 
-    config=`echo "$config " | sed -e 's/--with-[a-z]*= //g'`
+    config=$(echo "$config " | sed -e 's/--with-[a-z]*= //g')
+    config=$(echo "$config " | sed -e 's/--without-[a-z]* //g')
 
     if [ -n "$config" ]; then
         echo "\"$REPO_NAME\": the following config will be configure : \"$config\""
     fi
 
-    
-# create the configure script for we can configure it later
-    cat > $build/config.sh << EOF
+    if [ "${REPO_NAME}" = "ompi" ]; then
+        CONFIGURE_EXTRA_ARGS="${OMPI_CONFIGURE_EXTRA_ARGS}"
```

Comment:
`CONFIGURE_EXTRA_ARGS` and `OMPI_CONFIGURE_EXTRA_ARGS` this vars are not used, let's delete it.

Reply:
Used here

`# create the configure script for we can configure it later
    cat >"$build/config.sh" <<EOF
#!/bin/bash

echo "INFO: \$PATH"
echo "INFO: \$LD_LIBRARY_PATH"
echo "INFO: $REPO_SRC/configure ${CONFIGURE_EXTRA_ARGS} $config"

$REPO_SRC/configure ${CONFIGURE_EXTRA_ARGS} $config
EOF`

Reply:
Actually, the `OMPI_CONFIGURE_EXTRA_ARGS` is redundant variable, all extra args may be obtained thought `$config`